status-page.py

```diff
@@ -21,80 +23,39 @@ def __init__(self):
-            print(f'REQUEST ERROR! Unable to get incidents.\n{err}')
-            sys.exit(1)
+            response = self.session.get(f'{self.base_url}/incidents')
+        except RequestException as err:
+            sys.exit(f'REQUEST ERROR! Unable to get incidents.\n{err}')
         for incident in response.json()['data']:
-            status = incident['status']
-            if status != 4:
+            if incident['status'] != 4:
                 self.open_incidents.append(incident)
 
-    def update_incident_status(self):
-        """Goes through all passed and failed tests, resolves open incidents with a passed test, and creates new
-        incidents for failed tests."""
-        api_map = self.api_map
-
-        # If there are any test failures, open an incident if there is none open already
-        for failed_test in self.api_status_json['failed_tests']:
-            already_reported = False
-            parsed_url = urlparse(failed_test['base_url'])
-            split_path = parsed_url.path.split('/')
-            api_version = split_path[1]
-            api_name = split_path[2]
-            version_and_name = f'{api_version}/{api_name}'
-
-            if version_and_name in api_map:
-                component_id = api_map[version_and_name]
-                for incident in self.open_incidents:
-                    if component_id == incident['component_id']:
-                        already_reported = True
-                if not already_reported:
-                    body = {
-                        'name': failed_test['base_url'],
-                        'message': f'Jenkins has reported an issue with {api_name} api {api_version}.\
-                                     Failed test url: {failed_test["base_url"]}',
-                        'status': 1,
-                        'visible': 1,
-                        'component_id': component_id,
-                        'component_status': 2,
-
-                    }
-                    try:
-                        response = requests.post(f'{self.base_url}/incidents', headers=self.header, data=body)
-                    except requests.exceptions.RequestException as err:
-                        print(f'REQUEST ERROR! Incident not posted:\n{err}')
-                        sys.exit(1)
-                    print(f'Posted New Incident: {response}')
-
-        # then parse through passed tests and resolve any previously opened incidents if it shares the same endpoint
-        for passed_test in self.api_status_json['passed_tests']:
+    def parse_tests(self, passed):
+        """
+        Parses through a set of passed or failed tests. If the tests passed,
+        resolve any open incidents with a passed test. If the tests failed,
+        create new incidents for each failed test if not already open.
+        """
+        pass_or_fail = 'passed_tests' if passed else 'failed_tests'
+        for test in self.api_status_json[pass_or_fail]:
```

Comment:
Just my two cents, `test` is a little ambiguous for most of the cases. `test_case` probably is a better variable name for this case.